### DISCUSSION TITLE
Use password from secrets manager for MLFlow with basic auth enabled.

### DISCUSSION THREAD

**Discussion by janani2019** [mlflow/mlflow#14118] [Q&A]
Hi!
We have mlflow deployed on an EKS Cluster. 
These are our pod args 
```
 pod_args = [
    "server",
    "--backend-store-uri",
    "postgresql://${var.training_db_user}@${data.aws_db_instance.mlflow.endpoint}/${var.training_db_name}",
    "--artifacts-destination", "s3://${module.mlflow_training_bucket.bucket_name}",
    "--serve-artifacts",
    "--host", "0.0.0.0",
    "--port", var.app_container_port,
    "--app-name", "basic-auth"
  ]
```

I tried passing MLFLOW_TRACKING_USERNAME and MLFLOW_TRACKING_PASSWORD in pod args and also as environment variables. Did not work. 
Today, our mlflow app is using the default user name and password. But we want to have the password to be pulled from AWS secrets manager. Could you please help on how this can be achieved ?

TIA!

**Comment by bhushanasati25** [comment]
Hi! `MLFLOW_TRACKING_USERNAME` and `MLFLOW_TRACKING_PASSWORD` are easy to mix up here, but they are mainly for MLflow clients authenticating to the tracking server. They do not configure the server-side admin password for `--app-name basic-auth`.

For the basic-auth server, MLflow reads auth settings from `basic_auth.ini`. The clean way to customize that file is to mount your own config and point MLflow to it with `MLFLOW_AUTH_CONFIG_PATH`.

A minimal `basic_auth.ini` would look like this:

```ini
[mlflow]
default_permission = READ
database_uri = postgresql://user:password@host:5432/mlflow_auth
admin_username = admin
admin_password = <password-from-secret>
authorization_function = mlflow.server.auth:authenticate_request_basic_auth
```

In EKS, I would do it like this:

1. Store the admin password in AWS Secrets Manager.
2. Sync it into Kubernetes using External Secrets Operator or Secrets Store CSI Driver.
3. Mount or render a `basic_auth.ini` file into the MLflow pod.
4. Set `MLFLOW_AUTH_CONFIG_PATH` to that mounted file.
5. Start MLflow with `--app-name basic-auth`.

One important gotcha: `admin_password` is only used when the default admin user is created for the first time. If your auth database already has the `admin` user with the old/default password, changing `basic_auth.ini` may not rotate that existing user’s password. In that case, update the password through MLflow’s auth API/client, or intentionally recreate/migrate the auth database.

Also, `--backend-store-uri` is for MLflow tracking metadata. Basic-auth users and permissions use the `database_uri` inside `basic_auth.ini`, although that can also point to Postgres.

So the short version is: sync the Secrets Manager value into the pod, mount it into `basic_auth.ini`, set `MLFLOW_AUTH_CONFIG_PATH`, and make sure the auth DB is initialized or updated with that password.
